### DISCUSSION TITLE
Tag in Stack

### DISCUSSION THREAD

**Discussion by jemo07** [ForthHub/discussion#148] [Ideas]
Hi all, has anybody implemented a tags on a stack? 

https://godbolt.org/z/P6rMev397 

I'm trying to build a bare metal forth, ( which seemed allot easier at first than develop a new OS ) for a high availability set of use cases on x86 and AArch64 . 

More can be read here: 

https://github.com/jemo07/TolerantFORTH

**Comment by CCurl** [comment]
I love the idea of a BMF forth system!

Looking at your examples, I would think you'd want the tagging/inter-process messaging support to be FIFO, not LIFO.

**Reply by jemo07** [reply]
Why so? The way I envision it it’s to maybe have a parallel process, the stack would never be that deep and the process are mostly waiting on stack data to be consumed, you still need to manage order processing the stack, so DUP, SWAP, ROT can be used in these scenarios.  Maybe I’m over looking your intentions, can you explain?

**Reply by CCurl** [reply]
so that the receiving process can retrieve the messages in the order they were sent. that seems like a desirable behavior to me.

**Reply by jemo07** [reply]
Ah, I missed your reply, makes lot's of sense, I was actually making them serial stacks, meaning that no new stack values from a given tag, but yes, I can see why this would be a good behavior.

**Comment by CCurl** [comment]
You might have a look at the "Mouse" programming language. Its stack has complex objects, so I think it could support "tagged" messages.

http://techref.massmind.org/Techref/language/mouse/interpreter-c.htm

**Reply by jemo07** [reply]
Thank you again @CCurl
